### PR TITLE
Create new id privacy tab and move marketing prefs there

### PR DESCRIPTION
**Pull request by danmassie**
A number of new privacy / data settings are going to be made available to our users. We need to create a new home for these within edit profile.

Marketing preferences are currently at the bottom of the email preferences area. We should move these under a new tab within edit profile.

By moving these settings to the edit profile, we're forcing the user to re-authenticate.

Before:
screenshot from 2016-01-14 15 36 44

After:
upload_1_13_2016_at_11_23_30_am

upload_11_19_2015_at_2_53_21_pm

### CODE REVIEW

**Comment by paperboyo** [comment]
:gem:! Ela, Fergal and @johnduffell had a great hackday idea around that - well worth talking to them!

**Comment by danmassie** [comment]
@paperboyo Chris Wilk is the man to speak to. The identity team are about to implement a few features in this area.

**Comment by paperboyo** [comment]
Thanks, @danmassie. I'm just a matchmaker. And a blind one, too. Also, true privacy = best marketing, I think.

**Comment by johnduffell** [comment]
privacy and security are all our favourite things :+1: although our hackday project was a little more ambitious than could be done in a PR

**Comment by paperboyo** [comment]
What about marketing?! :wink:

No-no, of course, I just got excited and wanted to mention your work as always when this topic comes up, because I think it’s The Thing™. As always, unacceptably, I hijacked someone’s PR to do so. Sorry, @danmassie! :sweet_potato:

File: identity/app/views/fragments/profile/privacyForm.scala.html [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  idRequest: services.IdentityRequest,
+  user: com.gu.identity.model.User,
+  privacyForm: Form[form.PrivacyFormData])(implicit request: RequestHeader, messages: play.api.i18n.Messages)
+
+@import views.html.fragments.form.{checkboxField}
+@import form.IdFormHelpers._
+@import helper._
+
+
+<form class="form js-public-profile-form" novalidate action="@idUrlBuilder.buildUrl("/privacy/edit", idRequest)" role="main" method="post">
```
Comment (jamespamplin) [inline identity/app/views/fragments/profile/privacyForm.scala.html]: Do we have to build the URL in the view directly, or can we build it in the Controller (or elsewhere) instead?

File: identity/test/controllers/EditProfileControllerTest.scala [hunk 2]
```diff
@@ -88,6 +88,46 @@ class EditProfileControllerTest extends path.FreeSpec with ShouldMatchers with M
...
+          receiveGnmMarketing = Some(receiveGnmMarketing),
+          receive3rdPartyMarketing = Some(receive3rdPartyMarketing)
+        )
+      )
+      when(api.saveUser(Matchers.any[String], Matchers.any[UserUpdate], Matchers.any[Auth]))
+        .thenReturn(Future.successful(Right(updatedUser)))
+
+      val result = controller.submitPrivacyForm().apply(fakeRequest)
+
+      Await.result(result, 10.seconds)
```
Comment (jamespamplin) [inline identity/test/controllers/EditProfileControllerTest.scala]: Even though this probably wont fail, 10 seconds is quite long to wait, could this be just 1 second?

Comment (danmassie) [inline identity/test/controllers/EditProfileControllerTest.scala]: It is a long time. Its a copy and paste of the other other submit tests.

File: identity/app/views/fragments/profile/privacyForm.scala.html (hunk 1 above)
Comment (danmassie) [inline identity/app/views/fragments/profile/privacyForm.scala.html]: We should build it elsewhere , but I'm not refactoring all of the views as part of this. At the moment this copies the pattern used elsewhere.

Comment (jamespamplin) [inline identity/app/views/fragments/profile/privacyForm.scala.html]: Ok I understand. Keep it consistent for now, we can fix it properly later. Thanks.

**Comment by danmassie** [comment]
@jamespamplin timeout reduced

**Comment by jamespamplin** [comment]
:+1: looks good to me

**Pull request merged**